include/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+if (GINKGO_CHECK_CIRCULAR_DEPS)
+    add_library(ginkgo_public_api INTERFACE) # dummy target
+    set_property(TARGET ginkgo_public_api APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_SOURCE_DIR}")
+    set_property(TARGET ginkgo_public_api APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_BINARY_DIR}")
+    ginkgo_check_headers(ginkgo_public_api)
```

Comment:
This is just a comment, but I think this target can be useful at some point since that would allow for example to use it to create the main `ginkgo.hpp` file, or to install our headers in a specific location. It's a good thing to have that available now.